ipinfo/cmd_init.go

```diff
@@ -0,0 +1,301 @@
+		if err != nil {
+			return err
+		}
+		defer res.Body.Close()
+
+		// parse response.
+		rawBody, err := io.ReadAll(res.Body)
+		if err != nil {
+			return err
+		}
+		body := &signupCli{}
+		err = json.Unmarshal(rawBody, body)
+		if err != nil {
+			return err
+		}
+
+		err = openURL(body.SignupURL)
+		fmt.Println("If the link does not open, please go to this link to get your access token:")
+		fmt.Printf("%v\n", body.SignupURL)
+		fmt.Println("Press [Enter] when done if not automatically detected.")
+		if err != nil {
+			fmt.Println("Error opening link:", err)
+			return err
+		}
+
+		// Retrieving unique id from signup URL.
+		parsedUrl, err := url.Parse(body.SignupURL)
+		if err != nil {
+			fmt.Println("Error parsing URL:", err)
+			return err
+		}
+		cli_token := parsedUrl.Query().Get("cli_token")
+		if cli_token == "" {
+			fmt.Println("UID not found in URL")
+			return err
+		}
+
+		// Check if signup flow is completed.
+		maxAttempts := 200
+		count := 0
+		for {
+			count++
+			res, err := http.Get("https://ipinfo.io/signup/cli/check?cli_token=" + cli_token)
+			if err != nil {
+				return fmt.Errorf("%v", err)
+			}
+			defer res.Body.Close()
+
+			if res.StatusCode == http.StatusOK {
+				rawBody, err := io.ReadAll(res.Body)
+				if err != nil {
+					return err
+				}
+				body := &tokenCli{}
+				err = json.Unmarshal(rawBody, body)
+				if err != nil {
+					return err
+				}
+
+				// save token to file.
+				gConfig.Token = tok
+				if err := SaveConfig(gConfig); err != nil {
+					return err
+				}
+
+				fmt.Println("Account created successfully.")
+				break
+			}
+
+			if count == maxAttempts {
+				if _, err := fmt.Scanln(); err != nil {
+					return fmt.Errorf("%v", err)
+				}
+			}
+
+			time.Sleep(time.Second)
+		}
+	} else {
+		fmt.Println("Invalid input.")
+		return err
+	}
+
+	return nil
+}
+
+func checkValidity(tok string) error {
+	fmt.Println("checking token...")
+	tokenOk, err := isTokenValid(tok)
+	if err != nil {
+		return fmt.Errorf("could not confirm if token is valid: %w", err)
+	}
+	if !tokenOk {
+		return fmt.Errorf("invalid token")
+	}
+
+	return nil
+}
+
+func enterToken(tok string) (string, error) {
+	for tok == "" {
+		fmt.Printf("Enter token: ")
+		tokraw, err := term.ReadPassword(int(os.Stdin.Fd()))
+		fmt.Println()
+		if err != nil {
+			return "", err
+		}
+
+		tok = string(tokraw[:])
+
+		// exit if we have a token now.
+		if tok != "" {
+			break
+		}
+
+		fmt.Println("please enter a token")
+	}
+
+	return tok, nil
+}
+
+func isTokenValid(tok string) (bool, error) {
+	// make API req for true token validity.
+	res, err := http.Get("https://ipinfo.io/me?token=" + tok)
+	if err != nil {
+		return false, err
+	}
+	defer res.Body.Close()
+
+	// parse response.
+	me := &meResponse{}
+	if err := json.NewDecoder(res.Body).Decode(me); err != nil {
+		return false, err
+	}
+
+	// If no errors then me.Error should be empty
+	return me.Error == "", nil
+}
+
+func openURL(url string) error {
```

Comment:
I linked you some code to check which already handles this in a cross-platform way. We have a function from a library that's already installed, which does what's the correct way.

Reply:
@UmanShahzad which link you're referring to ?